### PR TITLE
Add ignore_errors to builder and options

### PR DESCRIPTION
**Pull request by ShayBox**
This solves #40

Technically enabling `--ignore-errors` is useless as that's the default behavior with yt-dlp, but not with youtube_dl, but this skips the exit code check, as any stdout will result in an exit code of 1.

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -146,7 +146,7 @@ pub struct JsonOutput {
 
 #[derive(Clone, Serialize, Deserialize, Debug, Default)]
 pub struct Playlist {
-    pub entries: Option<Vec<SingleVideo>>,
+    pub entries: Option<Vec<Option<SingleVideo>>>,
```
Comment (GyrosOfWar) [inline src/model.rs:149]: Is it necessary to have nested options here?

Comment (ShayBox) [reply src/model.rs:149]: Yes, if a video fails such as a "Premium" video, it puts "null" in the list of entries, which parses to `Option<SingleVideo>`

**Comment by ShayBox** [comment]
It also appears title can be null in flat playlists, very rarely
The case I found  
Playlist: `https://www.youtube.com/channel/UC6nSFpj9HTCZ5t-N3Rm3-HA`
Video(s): `4ZJWv6t-PfY and dacojrlGM7Q`

File: src/model.rs (hunk 1 above)
Comment (GyrosOfWar) [reply src/model.rs:149]: Hmm, I see. That's pretty awkward from an API perspective, unfortunately. I'll merge it for now and look into improvements before a new release. (also, sorry for taking so long)

Comment (ShayBox) [reply src/model.rs:149]: A custom serializer like the one in the file already might be able to filter out none from the entries, I'm not familiar with serde custom serializers

**Comment by ShayBox** [comment]
I figured out the custom deserializer, this solves the complex type issue :+1:

**Comment by GyrosOfWar** [comment]
Fantastic, thanks a lot!

**Pull request merged**